// SPDX-FileCopyrightText: Copyright (c) 2025 Cisco and/or its affiliates.
// SPDX-License-Identifier: Apache-2.0

syntax = "proto3";

package core.v1alpha1;

// A specific skills that an agent is capable of performing.
// Specs: https://schema.oasf.agntcy.org/skills.
//
// Mapping schema between OASF objects:
//  Key := {category_uid}/{class_uid} - from uids
//  Key := {category_name}/{class_name} - from names
//
// Example (https://schema.oasf.agntcy.org/classes/contextual_comprehension)
//  Key := 1/10101
//  Key := Natural Language Processing [1]/Contextual Comprehension [10101]
message Skill {
  // UID of the category.
  string category_uid = 1;
  // UID of the class.
  string class_uid = 2;
  // Name of the category.
  optional string category_name = 3;
  // Name of the class.
  optional string class_name = 4;
  // Additional metadata for this skill.
  map<string, string> annotations = 5;
  // Optional schema version.
  optional string version = 6;
}
